wagtail_live/models.py

```
@@ -59,6 +58,168 @@ class LivePageMixin(models.Model):
+            raise
+        return self.get_live_post_by_index(live_post_index)
+
+    def add_block_to_live_post(self, block_type, block, live_block):
+        """ Add a new content block to a live post.
+        
+        Args:
+            block_type (str):
+                Type of the block to add
+            block (Block):
+                Block to add to the live post.
+            live_block (LivePostBlock):
+                Live post in which the new block will be added.
+        """
+
+        live_block["content"].append((block_type, block))
+
+    def add_live_post(self, live_post, live_post_id):
+        """ Add a new live post to live page.
+
+        Live posts are added with a custom ID to facilitate
+        keeping their track. 
+        By default, the value of a live post's ID is
+        the timestamp of the corresponding message creation. 
+        A check is also done to ensure that live posts are always
+        sorted by time.
+
+        Args:
+            live_post (LivePostBlock):
+                live post to add
+            live_post_id (str):
+                Id of the new live_post
+        """
+
+        lp_index = len(self.live_posts) if self.live_posts else 0
+        post_id = float(live_post_id)
+        while self.live_posts and float(self.live_posts[lp_index - 1].id) > post_id:
+            lp_index -= 1
+
+        # Insert to keep posts sorted by time
+        self.live_posts.insert(lp_index, ("live_post", live_post, live_post_id))
+
+        self.last_update_at = now()
+        self.save()
+
+    def delete_live_post(self, live_post_id):
+        """Delete a live post by its ID.
+        
+        Args:
+            live_post_id (str): Id of the live post to delete.
+
+        Raises:
+            (someException) if live post with the given ID doesn't exist.
+        """
+
+        live_post_index = self.get_live_post_index(live_post_id=live_post_id)
+        if live_post_index == -1:
+            raise
```

This will never be reached I think, because self.get_live_post_index throws an error instead of returning -1.